### PR TITLE
mtail: 3.0.0-rc4 -> 3.0.0-rc34

### PR DESCRIPTION
**Pull request by bbigras**
###### Motivation for this change
`3.0.0-rc4` is over 2 years old. no release yet...

###### Things done

~~This update has a tiny lie.~~

~~I saw from the manual that `go2nix` is normally used for `goDeps` but on the `go2nix` repo it says to use `vgo2nix` when there a `go.mod` file present.~~

~~The thing is that `vgo2nix` fails with `3.0.0-rc34` but not with master, so I generated the file with master. master only has tiny changes. It's mostly a deps update. (check https://github.com/google/mtail/compare/v3.0.0-rc34...0150dadbe24cbb3cc09cd5859ce8632541fb235a)~~

~~If it's not OK I could another way. I also wanted to switch the package to buildGoModule but I got some problem: https://github.com/NixOS/nixpkgs/issues/79355~~

Also, is gotools needed?

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

/nix/store/z0dk0mhfsni2lpaqpw2pnjgs0y9vzsyl-mtail-3.0.0-rc4-bin	  40.4M
/nix/store/1m6f7yfly52fpa4rwwsvwqkrxmfd7h8a-mtail-3.0.0-rc35	  45.5M

### CODE REVIEW

**Comment by bbigras** [comment]
Another tiny thing is that `mtail --version` will print:
`mtail version 3.0.0-rc34 git revision invalid:-use-make-to-build go version go1.13.7 go arch amd64 go os linux`

since I only set `Version`. We could also set `Branch` and `Revision`.

https://github.com/google/mtail/blob/7b4270cb6e8b6ed1f2febe48e370cb68d38681f9/cmd/mtail/main.go#L77-L80

**Comment by bbigras** [comment]
It seems my problem with `buildGoModule` doesn't happen anymore. I made the change to use `buildGoModule` now.

File: pkgs/servers/monitoring/mtail/default.nix [hunk 1]
```diff
@@ -1,22 +1,25 @@
-{ lib, fetchFromGitHub, gotools, buildGoPackage }:
+{ lib, fetchFromGitHub, buildGoModule }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "mtail";
-  version = "3.0.0-rc4";
+  version = "3.0.0-rc34";
   goPackagePath = "github.com/google/mtail";
```
Comment (doronbehar) [inline pkgs/servers/monitoring/mtail/default.nix]: This is not needed with `buildGoModule`.

```suggestion
```

**Comment by bbigras** [comment]
done, thanks @doronbehar

File: pkgs/servers/monitoring/mtail/default.nix [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
-  version = "3.0.0-rc4";
-  goPackagePath = "github.com/google/mtail";
+  version = "3.0.0-rc34";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "mtail";
-    rev = "v${version}";
-    sha256 = "1n7pqvid48ayn15qfpgpbsx0iqg24x08wphzpc08mlfw47gq7jg3";
+    rev = "7b4270cb6e8b6ed1f2febe48e370cb68d38681f9";
```
Comment (doronbehar) [inline pkgs/servers/monitoring/mtail/default.nix]: Why don't you use https://github.com/google/mtail/releases ?

File: pkgs/servers/monitoring/mtail/default.nix [hunk 3]
```diff
@@ -1,22 +1,24 @@
-{ lib, fetchFromGitHub, gotools, buildGoPackage }:
+{ lib, fetchFromGitHub, buildGoModule }:
 
-buildGoPackage rec {
+buildGoModule rec {
```
Comment (doronbehar) [inline pkgs/servers/monitoring/mtail/default.nix]: If `buildGoModule` is used, why `deps.nix` still present?

File: pkgs/servers/monitoring/mtail/default.nix [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
   };
 
-  buildInputs = [ gotools ];
-  goDeps = ./deps.nix;
-  patches = [ ./fix-gopath.patch ];
-  preBuild = "go generate -x ./go/src/github.com/google/mtail/vm/";
+  modSha256 = "1iipmja3zcmpc5mg9q41p26wrccl9z937y40aqi0cyld7y94lafc";
+  subPackages = [ "cmd/mtail" ];
 
+  preBuild = "go generate -x ./internal/vm/";
```
Comment (doronbehar) [inline pkgs/servers/monitoring/mtail/default.nix]: I'd use a multiline nix string, for the sake of consistency across Nixpkgs:

```suggestion
  preBuild = ''
    go generate -x ./internal/vm/
  '';
```

**Comment by bbigras** [comment]
I done all the changes and updated to rc35. thanks

**Review by doronbehar — APPROVED** [review]
Great, tested basic `--help` and no comments left :).

**Pull request merged**